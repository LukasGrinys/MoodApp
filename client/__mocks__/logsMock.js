export const testLogs = [
    {
        id: '001',
        rating: 10,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '002',
        rating: 9,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '003',
        rating: 8,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '004',
        rating: 7,
        date: "TEST-DATE",
        text: "Test text"
    }
];

export const testLog = {
    id: '001',
    rating: 10,
    timing: 'Afternoon', 
    date: "TEST-DATE",
    text: "Test text"
}