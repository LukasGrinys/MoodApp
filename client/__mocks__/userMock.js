export const testUser = {
    id: '001',
    firstName: 'test',
    lastName: 'test',
    email: 'test@test.com'
}