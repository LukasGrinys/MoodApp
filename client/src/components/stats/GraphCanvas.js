import React, { Component } from 'react';
import styles from './stats.module.css';
import { graphStroke } from './../../widgets/nightmodeColors';

class GraphCanvas extends Component {
    state = {
        width: 0
    }
    constructor(props){
        super(props);
        this.myInput = React.createRef()
    };
    returnPolylinePoints = (list) => {
        let pointsString = '';
        const elementWidth = this.state.width;
        for (let i = 0; i < list.length; i++) {
            let ratingYpos = 220 - (list[i] * 20);
            pointsString += (Math.floor((elementWidth / 21) * (list.length - i - 1))) + "," + ratingYpos + " "; 
        }
        return pointsString
    }
    componentDidMount () {
        window.addEventListener('resize', () => {
            let newWidth = this.myInput.current.offsetWidth;
            if (newWidth !== this.state.width) {
                this.setState({
                    width: newWidth
                })
            };
        });
        if (this.state.width === 0) {
            this.setState({
                width: this.myInput.current.offsetWidth
            })
        };
    }

    render() {
        return (
            <div className={styles.graph_canvas} ref={this.myInput} >
                <div className={styles.y_axis_number}>10</div>
                <div className={styles.y_axis_number} style={{transform: "translate(10px, 100px)"}}>5</div>
                <div className={styles.y_axis_number} style={{transform: "translate(10px, 200px)"}}>0</div>
                <svg width={"100%"} height={"100%"}>
                    <polyline points={"1,20 490,20"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
                    <polyline points={"1,40 490,40"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
                    <polyline points={"1,60 490,60"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
                    <polyline points={"1,80 490,80"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
                    <polyline points={"1,100 490,100"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
                    <polyline points={"1,120 490,120"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
                    <polyline points={"1,140 490,140"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
                    <polyline points={"1,160 490,160"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
                    <polyline points={"1,180 490,180"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
                    <polyline points={"1,200 490,200"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
                    <polyline points={this.returnPolylinePoints(this.props.list)}
                    className={styles.graph_stroke} style={graphStroke(this.props.nightmode)}/>
                </svg>
            </div>
        )
    }
}

export default GraphCanvas;