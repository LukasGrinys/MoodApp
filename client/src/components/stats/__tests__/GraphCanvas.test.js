import React from 'react';
import GraphCanvas from '../GraphCanvas';
import { shallow } from 'enzyme';
import { testLogs } from '../../../../__mocks__/logsMock';

const wrapper = shallow(<GraphCanvas logs={testLogs}/>);

test('Renders properly', () => {
    expect(wrapper).toMatchSnapshot();
});