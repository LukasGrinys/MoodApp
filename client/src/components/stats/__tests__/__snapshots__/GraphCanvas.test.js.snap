// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<div
  className="graph_container"
>
  <p>
    <strong>
      Your mood graph
    </strong>
  </p>
  <div
    className="graph_canvas"
  >
    <div
      className="y_axis_number"
    >
      10
    </div>
    <div
      className="y_axis_number"
      style={
        Object {
          "transform": "translate(10px, 100px)",
        }
      }
    >
      5
    </div>
    <div
      className="y_axis_number"
      style={
        Object {
          "transform": "translate(10px, 200px)",
        }
      }
    >
      0
    </div>
    <svg
      height="100%"
      width="100%"
    >
      <polyline
        points="1,20 490,20"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,40 490,40"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,60 490,60"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,80 490,80"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,100 490,100"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,120 490,120"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,140 490,140"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,160 490,160"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,180 490,180"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,200 490,200"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <ReturnGraph
        points="0,20 0,40 0,60 0,80 "
      />
    </svg>
  </div>
  <div>
    From 
    TEST-DATE
     to 
    TEST-DATE
  </div>
</div>
`;
